### ISSUE TITLE
random number generation

### ISSUE THREAD

**Issue by Ung0d** [mslehre/MAS#34]
I just tested the sequence simulation code (at home under windows). **It does produce the same sequence every time I run the program. Also the sequences are not mutated at all.** I suspect that this is a seed initialization problem. Do you get different sequences for each execution of the program on your system? Do mutations work?

I researched a bit and found that std::random_device is deterministic under MINGW/Windows, but non-deterministic under linux, which is a ugly thing as we want code to be portable and work the same on every system.

As a solution I would try to not use std::random_device and set the seed using the current system time instead:

```
#include <chrono>

std::mt19937 rng(chrono::high_resolution_clock::now().time_since_epoch().count());
```

(and remove the line with the random device)

The issue with the mutations not working might be unrelated to this. Since you generate random doubles there, you should use `std::uniform_real_distribution`
(https://en.cppreference.com/w/cpp/numeric/random/uniform_real_distribution) for the random mutation probability, not the random device as you currently do.

**Comment by TimonKapischke** [comment]
I have just tested it under Linux. After every run you get a new sequence.

**Comment by TimonKapischke** [comment]
Exemple:
- CGGTTGGGCTGGCCCAAGGCCCAAAATTTC
- TTCTTCACCTGCTTGACCGAGTCTAAGCATG
- CTCCACTAGTCAATGTATACATGTCACTCCCC

**Comment by TimonKapischke** [comment]
The mutation works also under Linux: (Mutation probability = 1 (100%) )
>seq1
TGGCATGCCCCTGACTTATCTATATCGGCG
>seq2
AGAAGTTGCGGAAAACCGCGGGCCGCCTTA
>seq3
CATGCTCACCGTTCGGACGGCCGGTGCCCG

**Comment by TimonKapischke** [comment]
I made a pull request with the new seed simulation. I hope it works under windows.

**Comment by Ung0d** [comment]
Fixed in #35
